File: Arrays/Algorithms/Rotation/array_rotation_left.cpp
#include <bits/stdc++.h>
using namespace std;

void leftRotate(vector<int>& arr, int d) {
	int n = arr.size();
	d = d % n;
	reverse(arr.begin(), arr.begin() + d);
	reverse(arr.begin() + d, arr.end());
	reverse(arr.begin(), arr.end());
}

int main() {
	int n, d;
	cin >> n >> d;
	vector<int> arr(n);
	for (int i = 0; i < n; i++) cin >> arr[i];
	leftRotate(arr, d);
	for (int x : arr) cout << x << ' ';
	cout << endl;
	return 0;
}

File: Arrays/Algorithms/Rotation/array_rotation_left.py
def left_rotate(arr, d):
	n = len(arr)
	d = d % n
	arr[:] = arr[d:] + arr[:d]

if __name__ == "__main__":
	n, d = map(int, input().split())
	arr = list(map(int, input().split()))
	left_rotate(arr, d)
	print(*arr)
